moveit_planners/trajopt/include/trajopt_interface/trajopt_interface.h

```diff
@@ -0,0 +1,81 @@
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of PickNik nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+/* Author: Omid Heidari */
+
+#ifndef TRAJOPT_INTERFACE_TRAJOPT_INTERFACE_H
+#define TRAJOPT_INTERFACE_TRAJOPT_INTERFACE_H
+
+#include <ros/ros.h>
+
+#include <trajopt_sco/sco_common.hpp>
+
+#include "problem_description.h"
+
+#include <moveit/planning_interface/planning_interface.h>
+
+namespace trajopt_interface
+{
+MOVEIT_CLASS_FORWARD(TrajOptInterface);
+
+class TrajOptInterface
+{
+public:
+  TrajOptInterface(const ros::NodeHandle& nh = ros::NodeHandle("~"));
+
+  const sco::BasicTrustRegionSQPParameters& getParams() const
+  {
+    return params_;
+  }
+
+  bool solve(const planning_scene::PlanningSceneConstPtr& planning_scene,
+             const planning_interface::MotionPlanRequest& req, moveit_msgs::MotionPlanDetailedResponse& res);
+
+protected:
+  /** @brief Configure everything using the param server */
+  void setTrajOptParams(sco::BasicTrustRegionSQPParameters& param);
+  void setDefaultTrajOPtParams();
+  void setProblemInfoParam(ProblemInfo& problem_info);
+  void setJointPoseTermInfoParams(JointPoseTermInfoPtr& jp, std::string name);
+
+  ros::NodeHandle nh_;  /// The ROS node handle
+  sco::BasicTrustRegionSQPParameters params_;
+  std::vector<sco::Optimizer::Callback> callbacks_;
+  TrajOptProblemPtr prob_;
+};
+
+void callBackFunc(sco::OptProb* opt_prob, sco::OptResults& opt_res);
```

Comment:
remove?

Reply:
Left it for later development but we can remove it or I can add a TODO

Reply:
add TODO on why its empty